### PR TITLE
Layers Panel: Layer Action Scrim Container

### LINKED ISSUE #9140: Layers Panel: Layer Action Scrim Container
## Description

Create a container for the layer actions that appears on hover. This container should have a scrim to the left of it that overlays the title string when it's visible.

~Instead of an ellipsis on long layer titles, string should fade out (this will work better with the new delete, duplicate buttons that are getting introduced). The gradient would appear over text only in the hover state/when icons are on the panel—if the text is too long for the panel, it will still trail off with an ellipsis.~

Figma: https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor-Sprint-Feature-Level-Designs_?node-id=9954%3A422464
Designs as of Oct 4, 2021

Screen Shot 2021-10-04 at 11 38 09 AM
Screen Shot 2021-10-04 at 11 38 21 AM

## To Do
- [x] - Final UX from design team

## Acceptance Criteria 

- visual update as defined in figma

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Part of design updates for layers panel.

## Summary
Adds a container that will house the layer actions (soon to be built). This container appears on hover. Right now there are no actions other than the disabled `lock/unlock` action on the Background element.

## Relevant Technical Choices
Went with a css predominant approach to minimize having to hold button state in js.

Also went with a box-shadow implementation for the scrim instead of an element with a gradient to simplify positioning and nodes.

## User-facing changes
NA (Small right aligned scrim appears on hover in the layers panel now)

## Testing Instructions
Interact with the layers panel and see that there's now a little right aligned scrim that appears on hover. Other than that, just make sure nothing had any regressions.

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9140

### CODE REVIEW

File: packages/story-editor/src/components/panels/design/layer/layer.js [hunk 1]
```diff
@@ -112,29 +206,43 @@ function Layer({ layer }) {
...
+            <IconWrapper>
+              <Icons.LockClosed />
+            </IconWrapper>
           )}
-        </LayerContentContainer>
+        </LayerDescription>
+      </LayerButton>
+      <ActionsContainer>
+        {/*
+         *@TODO add layer actions here. only reason we conditionally
```
Comment (BrittanyIRL) [inline packages/story-editor/src/components/panels/design/layer/layer.js]:
```suggestion
         *@TODO #9137 #9138 add layer actions here. only reason we conditionally
```

**Review by BrittanyIRL — APPROVED** [review]
Well this CSS was fun to read through 😻  

Thanks for the notes!

File: packages/story-editor/src/components/panels/design/layer/layer.js [hunk 2]
```diff
@@ -30,41 +31,79 @@ import { LayerText } from '../../../../elements/shared/layerText';
...
+const ActionsContainer = styled.div`
+  position: absolute;
+  display: none;
+  align-items: center;
+  height: 100%;
+  top: 0;
+  right: 0;
+  padding-right: 6px;
+
+  --background-color: ${({ theme }) =>
```
Comment (samwhale) [inline packages/story-editor/src/components/panels/design/layer/layer.js:43]: oh sick - I didn't realize we could use css variables

**Review by samwhale — APPROVED** [review]
Dope, very edumacational

File: packages/story-editor/src/components/panels/design/layer/layer.js [hunk 3]
```diff
@@ -31,41 +31,79 @@ import { LayerText } from '../../../../elements/shared/layerText';
...
+  box-shadow: 0px 0px 15px 20px var(--background-color);
+`;
+
+const LayerContainer = styled.div`
+  position: relative;
+  height: ${LAYER_HEIGHT}px;
+  width: 100%;
+  overflow: hidden;
+
+  :is(:hover, :focus-within) ${ActionsContainer} {
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/panels/design/layer/layer.js:55]: 🤯 love it.

**Review by BrookeGraham — APPROVED** [review]
🙌🏽

**Pull request merged**